### PR TITLE
Return composed errors only

### PR DESCRIPTION
**Pull request by tom-james-watson**
* Refactor so that all errors raised through ajv are composed via structure-composer-error.
* Allow custom keywords to be added
* Tests

### CODE REVIEW

**Comment by tom-james-watson** [comment]
You'll want to look at the two commits separately.

You'll also need this version of structure-error-codes for the tests to pass https://github.com/f12/structure-error-codes/blob/feature/errorCodes/index.js

**Comment by tom-james-watson** [comment]
Also, I've just realised I'm not sure how this will support custom keywords. Will put some thought to that when I'm back

File: src/ajv-schema.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  /*
+   * Handle ajv unhandled errors
+   *
+   * @private
+   * @param {Object} error - ajv unhandled error
+   * @returns {Object} - structured unhandled error
+   */
+  handleUnhandled(error) {
+    return {
+      errorCode: 'UNHANDLED_SCHEMA_REJECTION',
```
Comment (chrisabrams) [inline src/ajv-schema.js]: Change to: `SCHEMA_REJECTION_UNHANDLED`

File: src/ajv-schema.js [hunk 2]
```diff
@@ -70,7 +86,7 @@ export default class AjvSchema {
 
       return errorComposer.compose(errorCode, error, {status: statusCode})
 
-    });
+    })
```
Comment (chrisabrams) [inline src/ajv-schema.js]: lol

Comment (tom-james-watson) [inline src/ajv-schema.js]: 😆

**Pull request merged**